app/serializers/notification_serializer.rb

```diff
@@ -1,23 +1,20 @@
 # frozen_string_literal: true
 
 # NB: at the moment this serializer only supports Move topics. In the future we should also support Person topics.
-class NotificationSerializer < ActiveModel::Serializer
-  include Rails.application.routes.url_helpers
+class NotificationSerializer
+  include JSONAPI::Serializer
+
+  set_type :notifications
 
   attributes :id, :event_type
-  attribute :created_at, key: :timestamp
 
-  # this is a little ugly but it is the only way (?!) to get the serializer to behave exactly as required
-  belongs_to :topic, polymorphic: true, key: 'move', if: :move? do |serializer|
-    link :self, serializer.move_url
-    object.topic
-  end
+  attribute :timestamp, &:created_at
 
-  def move?
-    object.topic.is_a?(Move)
-  end
+  belongs_to :topic, polymorphic: true, key: :move, if: proc { |object| move?(object) }, links: {
```

Comment:
Smells like Notifications don't need to have a polymorphic topic. Could you add a method to the `Notification` model that points at the topic:

```ruby
def move
  topic if topic.is_a?(Move)
end
```
 or equivalent and tidy this up a bit? I don't think we need the test of move even since I'm guessing that jsonapi serializer doesn't try to serialize nil?

Reply:
Interesting idea, I'll have a look

Reply:
I tried this but it didn't gain much - the `proc` is still needed to handle the case of an optional `move` relationship so it ends up polluting the model with serializer concerns. I think I'll leave the existing implementation instead.

Reply:
Thanks for trying :)